crypto/pkcs11key/key.go

```diff
@@ -252,10 +312,117 @@ func getPublicKey(module ctx, session pkcs11.SessionHandle, privateKeyHandle pkc
 	if !gotModulus || !gotExponent {
 		return noKey, errors.New("public key missing either modulus or exponent")
 	}
-	return rsa.PublicKey{
+
+	rsa := rsa.PublicKey{
 		N: n,
 		E: e,
-	}, nil
+	}
+	return rsa, nil
+}
+
+// Get the public key matching an Elliptic Curve private key
+func (ps *Key) getECPublicKey(module ctx, session pkcs11.SessionHandle, privateKeyHandle pkcs11.ObjectHandle) (interface{}, error) {
```

Comment:
This function is a bit on the long side and would probably benefit from being split into two, perhaps around `publicKeyHandle := objs[0]`.